components/image/slider/src/index.js

```diff
@@ -1,45 +1,73 @@
+import React, {Component} from 'react'
 import PropTypes from 'prop-types'
-import React from 'react'
 import ReactSlidy from 'react-slidy'
+import cloneDeep from 'lodash.clonedeep'
+import cx from 'classnames'
+// TODO: To be used when we can publish packages again to the npm registry.
+// import IconCamera from '@schibstedspain/sui-svgiconset/lib/Camera'
 
-const ImageSlider = (props) => {
-  const slides = getSlides(props.images, props.linkFactory)
+const NO_OP = () => {}
 
-  return (
-    <div onClick={props.handleClick} className='sui-ImageSlider'>
-      { hasMoreThanOneImage(props.images)
-        ? (<ReactSlidy {...props.sliderOptions} dynamicContent={props.dynamicContent}>{ slides }</ReactSlidy>)
-        : slides
-      }
-    </div>
-  )
-}
+class ImageSlider extends Component {
+  constructor (props) {
+    super(props)
+    const {sliderOptions} = props
+    // In order to accept a custom doAfterSlide callback, and to avoid altering props.sliderOptions
+    // sliderOptions is deeply cloned.
+    this._sliderOptions = cloneDeep(sliderOptions)
+    this._sliderOptions.doAfterSlide = (currentSlide) => {
+      this.setState(currentSlide)
+      sliderOptions.doAfterSlide && sliderOptions.doAfterSlide(currentSlide)
+    }
+  }
+
+  state = {currentSlide: 0}
+
+  render () {
+    const {images, linkFactory, handleClick, dynamicContent, enableCounter} = this.props
+    const slides = this._getSlides(images, linkFactory)
```

Comment:
This could be: `const slides = images && images.length ? this._getSlides(images, linkFactory) : []`

Reply:
Regarding this comment see https://github.com/SUI-Components/sui-components/pull/311/files#r166914428